lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3370,7 +3370,7 @@ namespace {
 
     /// Emit diagnostics for incorrect usage of SWIFT_RETURNS_RETAINED and
     /// SWIFT_RETURNS_UNRETAINED
-    void checkBridgingAttrs(const clang::FunctionDecl *decl) {
+    void checkBridgingAttrs(const clang::NamedDecl *decl) {
```

Comment:
It only makes sense to call this with `FunctionDecl` or `ObjCMethodDecl` right? Let's add an assertion that is triggered if it is called with a different type. Alternatively, you could use a pointer union of `FunctionDecl` and `ObjCMethodDecl`.